caffe2/ideep/operators/fully_connected_op.cc

```diff
@@ -2,67 +2,111 @@
 
 namespace caffe2 {
 
+USE_IDEEP_DEF_ALIASES();
+
+static inline itensor::dims CanonicalAxis(itensor::dims adims, size_t axis) {
+  if (adims.size() == 2 || axis == 1)
+    return adims;
+  auto dim0 = std::accumulate(adims.begin(), adims.begin() + axis, 1,
+                              std::multiplies<itensor::dim_t>());
+  auto dim1 = std::accumulate(adims.begin() + axis, adims.end(), 1,
+                              std::multiplies<itensor::dim_t>());
+  return itensor::dims({dim0, dim1});
+}
+
 class IDEEPFullyConnectedOp final : public IDEEPOperator {
- public:
+public:
   USE_IDEEP_DEF_ALIASES();
   USE_IDEEP_OPERATOR_FUNCTIONS();
 
-  IDEEPFullyConnectedOp(const OperatorDef& operator_def, Workspace* ws)
+  IDEEPFullyConnectedOp(const OperatorDef &operator_def, Workspace *ws)
       : IDEEPOperator(operator_def, ws),
+        axis_(OperatorBase::GetSingleArgument<int32_t>("axis", 1)),
+        axis_w_(OperatorBase::GetSingleArgument<int32_t>("axis_w", 1)),
         float16_compute_(
             OperatorBase::GetSingleArgument<bool>("float16_compute", false)) {}
   virtual ~IDEEPFullyConnectedOp() {}
 
   bool RunOnDevice() override {
-    const auto& X = Input(INPUT);
-    const auto& filter = Input(FILTER);
-    auto* Y = Output(OUTPUT);
+    const auto &X = Input(INPUT);
```

Comment:
Can we not change the format back to `auto &X`?  Our coding style is `auto& X`.

Reply:
will fix it soon